packages/config-loader/src/lib/schema/compile.ts

```diff
@@ -126,6 +128,11 @@ export function compileConfigSchemas(
       };
     }
 
+    // Given we know this is a valid config, re-validate each config's data
+    // values individually in order to allow type coercion mutations to make
+    // their way up to callers.
+    configs.forEach(conf => validate(conf.data));
```

Comment:
I don't think we know that this is all valid data at this point, since the merged config above might be hiding invalid data that was corrected in a later config file.

This mutated the underlying data right? I'm not very keen on missing data under the hood, it would be cleaner to return the mutated data instead. It's too high risk that the mutation has unforseen consequences. 

We'd need to make sure this work for backend as well in an explicit way. Right now I assume it may work most of the time because of the secret extraction, but it's not a given that that always happens.

In general I feel it would be best to make schema processing something that is always part of config loading if we go down this route. Which I don't think would be that bad

Reply:
I'm realizing that this mutation is actually quite unnecessary (assuming it happens). We can just have the type coercion for the schema check and then throw it away and rely on the coercion in `getBoolean` instead at runtime?

With my assumption of how this works I'm thinking that we could simply skip this line and :shipit:?

Reply:
Could do, but if so, we'd have to add similar coercion/handling to `getNumber`.  ...It would also introduce a (probably very rare) situation where someone could provide boolean/numeric values for string configs that would pass validation, but then cause errors at run-time (unless we also added coercion/handling to `getString`).

So I could either go down your initial line of thinking, or clean this up and add the extra coercion handling at read-time.  Whaddya think @Rugvip?

Reply:
I'm leaning towards adding additional coercion to the reader tbh, probably just `getNumber`. I feel we've already set that precedent and that it adds the lead amount of complexity. In particular it keeps the runtime config reading fairly decoupled from the schema still, which I think is desierable to the extent that it's possible.

In particular adding config schema processing to the backend would be a larger change which could have some impact. If we do coercion based on schema it becomes a lot more important that all schema is available at runtime, which isn't a given right now.

Reply:
Turns out there was already coercion going on in `getNumber`.

Addressed in the latest version.  Have one last look @Rugvip?